### PR TITLE
[Common] Support GRAY8, the 8-bit grayscale video format - @open sesame 09/03 16:57

### PR DESCRIPTION
**Pull request by wooksong**
Relates: #74 

**Changes proposed in this PR:**
- Adding support for the 8-bit graycale video format, GRAY8
- Modifying tools (such as bmp2png, gen24bBMP.py, and testAPI.sh) in the test framework to support the newly added color format, GRAY8
- Adding test cases for the 8-bit graycale video format (i.e., video/x-raw,format=GRAY8)

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

Signed-off-by: Wook Song <wook16.song@samsung.com>

### CODE REVIEW

File: common/tensor_common.c [hunk 1]
```diff
@@ -449,6 +449,9 @@ gst_tensor_config_from_tensor_structure (GstTensorConfig * config,
   if (config->tensor_media_type == _NNS_VIDEO) {
     if (config->type == _NNS_UINT8) {
       switch (config->dimension[0]) {
+        case 1:
+          config->tensor_media_format = GST_VIDEO_FORMAT_GRAY8;
+          break;
```
[the remark this answers is not in the document]
Comment (jaeyun-jung) [reply common/tensor_common.c]: Now I am preparing refactored plugin 'tensorconverter' and 'tensordec' to remove the media type, format and rank in tensor config.

File: tests/testAPI.sh [hunk 2]
```diff
@@ -210,6 +210,10 @@ function convertBMP2PNG {
 	fi
 	for X in `ls *.bmp`
 	do
-		$tool $X
+        if [[ $X  = *"GRAY8"* ]]; then
```
Comment (myungjoo) [inline tests/testAPI.sh]: Fix indentation please.

**Comment by myungjoo** [comment]
Other than the indentation issue above, LGTM.

File: tests/testAPI.sh (hunk 2 above)
Comment (wooksong) [reply tests/testAPI.sh]: Done.

File: tests/testAPI.sh [hunk 3]
```diff
@@ -210,6 +210,10 @@ function convertBMP2PNG {
 	fi
 	for X in `ls *.bmp`
 	do
-		$tool $X
+                if [[ $X  = *"GRAY8"* ]]; then
+                        $tool $X --GRAY8
```
Comment (wooksong) [inline tests/testAPI.sh]: The indentation has been not fixed yet. Weird...

**Comment by wooksong** [comment]
The indentation issue is fixed.

**Pull request merged**